velox/functions/sparksql/Comparisons.cpp

```diff
@@ -37,28 +38,102 @@ class ComparisonFunction final : public exec::VectorFunction {
       const TypePtr& outputType,
       exec::EvalCtx& context,
       VectorPtr& result) const override {
+    const Cmp cmp;
     context.ensureWritable(rows, BOOLEAN(), result);
+    result->clearNulls(rows);
+    vector_size_t size = rows.end() - rows.begin();
+
+    if constexpr (
+        kind == TypeKind::TINYINT || kind == TypeKind::SMALLINT ||
+        kind == TypeKind::INTEGER || kind == TypeKind::BIGINT) {
+      if ((args[0]->isFlatEncoding() || args[0]->isConstantEncoding()) &&
+          (args[1]->isFlatEncoding() || args[1]->isConstantEncoding()) &&
+          rows.isAllSelected()) {
+        applySimdComparison<T, Cmp>(rows, args, result);
+        return;
+      }
+    }
+
+    if (size > 32) {
+      if (args[0]->isFlatEncoding() && args[1]->isFlatEncoding()) {
```

Comment:
The dispatch on encodings can also be extracted to `SIMDComparisonUtil.h`

Reply:
Updated, thanks.